src/build_loss_objective.jl

```diff
@@ -83,15 +76,17 @@ function build_loss_objective(prob::SciMLBase.AbstractSciMLProblem, alg, loss,
     elseif adjsa_gradient
         g! = (x, out) -> diffeq_sen_l2!(out, prob.f, prob.u0, prob.tspan, x, loss.t,
                                         loss.data, alg)
-    else
-        g! = (x, out) -> Calculus.finite_difference!(cost_function, x, out, :central)
     end
 
-    cost_function2 = function (p, grad)
-        if length(grad) > 0
-            g!(p, grad)
+    if flsa_gradient || adjsa_gradient
+        cost_function2 = function (p, grad)
+            if length(grad) > 0
+                g!(p, grad)
+            end
+            cost_function(p)
         end
-        cost_function(p)
+    else
+        cost_function2 = nothing
```

Comment:
the cost_function2 can just in general be removed.

Reply:
Yeah even I thought that but wasn't sure if it would be too breaking

Reply:
Go for broke: This should get a major release with this change anyways. Just make these functions that return objective functions into functions that return an OptimizationProblem or something of the sort.

Reply:
Okay cool